mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -119,6 +119,18 @@ final class EMLDAOptimizer extends LDAOptimizer {
       }
     }
 
+    // create term vertices for empty docs
+    val emptyDocTermVertices: RDD[(VertexId, TopicCounts)] = {
+      //empty documents
+      val emptyDocs = docs.filter { case (d: Long, tc: Vector) =>
+        tc.toBreeze.reduce(_+_) != BDV(0.0)
+      }
+      emptyDocs.map { case(docID: Long, termCounts: Vector) =>
```

Comment:
You can chain this to avoid more declarations, and seems like you can just use `mapValues`? and would `Vectors.zeros` be simpler and easier?